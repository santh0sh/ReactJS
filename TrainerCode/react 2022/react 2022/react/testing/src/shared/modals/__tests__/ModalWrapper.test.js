import { shallow, mount } from 'enzyme';
import ModalWrapper from '../ModalWrapper';

describe('Render ModalWrapper', () => {
    it('without component', () => {
        const ModalWrapperComponent = shallow(<ModalWrapper />);
        expect(ModalWrapperComponent).toMatchSnapshot();
    });

    it('with component', () => {
        const props = {
                component: () => {}
            },
            ModalWrapperComponent = shallow(<ModalWrapper {...props} />);
        expect(ModalWrapperComponent).toMatchSnapshot();
    });

    it('render correct class name', () => {
        const props = {
                modalClassName: 'custom-class-name'
            },
            ModalWrapperComponent = shallow(<ModalWrapper {...props} />).find('Modal');
        expect(ModalWrapperComponent.hasClass('custom-class-name')).toEqual(true);
    });

    it('render correct title', () => {
        const props = {
                title: 'Modal Title'
            },
            ModalWrapperComponent = shallow(<ModalWrapper {...props} />).find('ModalTitle');
        expect(ModalWrapperComponent.props().children).toEqual('Modal Title');
    });

    describe('render correct show prop', () => {
        const props = {
                show: true
            },
            ModalWrapperComponent = shallow(<ModalWrapper {...props} />).find('Modal');

        it('check prop type', () => {
            expect(ModalWrapperComponent.props().show).toBeBoolean();
        });

        it('check prop value', () => {
            expect(ModalWrapperComponent.props().show).toEqual(true);
        });
    });

    it('render correct onHide prop type', () => {
        const props = {
                onHide: () => {}
            },
            ModalWrapperComponent = shallow(<ModalWrapper {...props} />).find('Modal');
        expect(ModalWrapperComponent.props().onHide).toBeFunction();
    });

    it('render correct component prop type', () => {
        const props = {
                component: () => {}
            },
            ModalWrapperComponent = mount(<ModalWrapper {...props} />);
        expect(ModalWrapperComponent.props().component).toBeFunction();
    });
});
