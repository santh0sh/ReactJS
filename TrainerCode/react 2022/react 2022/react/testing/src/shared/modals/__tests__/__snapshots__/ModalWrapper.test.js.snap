// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ModalWrapper with component 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={false}
>
  <div
    className="modal-container"
  >
    <ModalHeader
      bsClass="modal-header"
      closeButton={false}
      closeLabel="Close"
    >
      <ModalTitle
        bsClass="modal-title"
        componentClass="h4"
      />
      <div
        className="close-btn"
      >
        ✕
      </div>
    </ModalHeader>
    <ModalBody
      bsClass="modal-body"
      componentClass="div"
    >
      <component
        component={[Function]}
      />
    </ModalBody>
  </div>
</Modal>
`;

exports[`Render ModalWrapper without component 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={false}
>
  <div
    className="modal-container"
  >
    <ModalHeader
      bsClass="modal-header"
      closeButton={false}
      closeLabel="Close"
    >
      <ModalTitle
        bsClass="modal-title"
        componentClass="h4"
      />
      <div
        className="close-btn"
      >
        ✕
      </div>
    </ModalHeader>
    <ModalBody
      bsClass="modal-body"
      componentClass="div"
    />
  </div>
</Modal>
`;
